### ISSUE TITLE
upgrade in offline mode to another PC F37->F38

### ISSUE THREAD

**Issue by semaka** [rpm-software-management/dnf-plugin-system-upgrade#61]
There is a magic plugin "dnf-plugin-system-upgrade" which has magic commands but still can't see differences in using commands system-upgrade and offline-upgrade.
I have my Fedora37 OS customized to my preferences then I do the steps:

```
#dnf update
#reboot
#dnf system-upgrade download --releasever=38
#dnf system-upgrade reboot
```
The upgrade is done well when rebooting. The following commands give me same results and I cant see the differences
```
#dnf update
#reboot
#dnf offline-upgrade download --releasever=38
#dnf offline-upgrade reboot
```

The first question is: what is the difference between the two procedure for upgrade as soon as I get same results?

But the main problem is here in following procedure.
```
#dnf update
#reboot
#dnf offline-upgrade download --releasever=38
```
- at this step I get the entire folder **/var/lib/dnf** with all my downloaded packages and replace it to another PC which is in offline then I give the next command.
`
#dnf offline-upgrade reboot`

- the system is rebooting but not in upgrading mode, give some errors then booting without upgrade. Is it something missed somewhere else than dnf folder?

So the question is, never found on internet, how to upgrade in offline mode replicated to another PC not the one which is in online. What something else is missing as soon as I have all packages in dnf folder.

**Comment by j-mracek** [comment]
The `system-upgrade` is very similar to `offline-distrosync`. Both commands can downgrade packages if only lover version is available. The `system-upgrade` has check whether releasever value is higher then the current value on the system. Additionally the `system-upgrade` might modify enabled stream, apply modules obsoletes and upgrate comps groups.

**Comment by semaka** [comment]
> The `system-upgrade` is very similar to `offline-distrosync`. Both commands can downgrade packages if only lover version is available. The `system-upgrade` has check whether releasever value is higher then the current value on the system. Additionally the `system-upgrade` might modify enabled stream, apply modules obsoletes and upgrate comps groups.

good to know as info but I still not have a solution to my problem. How to perform download to a PC and upgrade to another by moving just downloaded files??? The question is what the online PC is doing during download in such way that I can perform upgrade but this can't be done on another PC where just moved the folder but not download done.

**Comment by j-mracek** [comment]
May I ask you for the error that it provided during the reboot.

**Comment by j-mracek** [comment]
Personally I would be careful with replacing `/var/lib/dnf`. My suggestion is to only copy `/var/lib/dnf/system-upgrade`.
It contains history DB and replacing it might create a mess.

**Comment by j-mracek** [comment]
I believe that the problem is in PGP key verification, because keys are imported during performing download command. This could be verified by adding `--nogpgcheck` to `dnf system-upgrade download --releasever=38`.
Additionally there might a difference in installed packages or versions.

**Comment by semaka** [comment]
> Personally I would be careful with replacing `/var/lib/dnf`. My suggestion is to only copy `/var/lib/dnf/system-upgrade`. It contains history DB and replacing it might create a mess.

I don't think this will be different as soon as the three files under dnf are like originator, but I will do as you suggested, just system-upgrade. Regarding your next answer I think I'm close to find the problem and it is in the way indicated by you "gpgcheck". I will try your suggestion and come back with an answer. Thank you for help.

**Comment by semaka** [comment]
Sorry but it doesn't work even if I downloaded the packages without gpgcheck. In documents of the plugin somewhere is written that "dnf system-upgrade reboot" is depended of "dnf system-upgrade download...", maybe in system-upgrade download something is configured somewhere to be taken into account at next reboot. 

This is why my new PC which has just packages copied but no download command, does not reboot in upgrading mode. The creator of this plugin has any idea? See attached the error lines at rebooting moment.
![upgradeF37_error](https://github.com/rpm-software-management/dnf-plugin-system-upgrade/assets/6611018/378a7245-b71f-43a8-a90c-e65316c14fd3)

**Comment by j-mracek** [comment]
May I ask you for a log.
`dnf system-upgrade log`
Then use the last event number and export it to a file.
`dnf system-upgrade log --number=<latest_usually_1> > text.txt`

**Comment by semaka** [comment]
I solved the problem in another way. It is clear that "system-upgrade download" generate something important for "system-upgrade reboot", as it is mentioned in doc of plugin that reboot can't be done without download.
So I prepared the downloaded packages to be ready and available for a new local repo created. So I created a repository locally for this folder with rpms and I launched "system-ugrade download" in offline mode just with my downloaded packages through this local repo. Then I introduced "system-upgrade reboot" and magic, it's working. 
I don't think I need to investigate more to find why not working with dnf management.
